benchmarks/definition/config.py

```diff
@@ -0,0 +1,43 @@
+"""Benchmark configurations."""
+
+from enum import Enum, auto
+from typing import TypeVar
+
+from attrs import define, field
+from attrs.validators import deep_mapping, instance_of
+
+from baybe.recommenders.base import RecommenderProtocol
+from baybe.serialization.mixin import SerialMixin
+
+BenchmarkConfig = TypeVar("BenchmarkConfig")
+
+
+class _DEFAULT(Enum):
+    NOTHING = auto()
+
+
+# Sentinel indicating the absence of a recommender specification
+DEFAULT_RECOMMENDER = _DEFAULT.NOTHING
+
+
+@define(frozen=True)
+class RecommenderConvergenceAnalysis(SerialMixin):
```

Comment:
not sure if I fully get this class, why is it called `Analysis` ? this specifies settings / a config, where is any `analysis` done? The name suggests that `analysis` is somehow part of this, but this class is rather supposed to be sued for analysis  suppose, so seems like a misnomer to me

Reply:
Ok, I see that the `Analysis` part can be misleading. My rational was something like it is a `RecommenderConvergenceAnalysisConfig` but the config seemed a bit redundant when all these classes would end with it. But I agree that we should find a better name.

`MonteCarloExperiment` on the other hand seems way too generic to me, it doesn't say anything about the fact that we do closed-loop convergence tests – it could also mean anything else MC-related. So I suggest something that conveys the exact purpose/type of the benchmark... Assuming we keep it specific to recommenders [(see other thread)](https://github.com/emdgroup/baybe/pull/413#discussion_r1831348830), how about something like some combination of words like `Recommender` / `Convergence` / `ClosedLoop` / `Comparison` / `Benchmark` / `Config` / ...? Example for campaign-specific setting: `CampaignConvergenceBenchmarkConfig`